src/components/Advanced/TextBox/index.js

```diff
@@ -50,10 +50,13 @@ class TextBox extends Component {
     // Change ast
     if (!_.isEqual(prevProps.ast, this.props.ast)) {
       const translation = translate_raw(this.props.ast)
-      this.setState({
-        raw: translation || '',
-        edit: Object.keys(this.props.ast).length === 0
-      })
+      let isInsert = prevProps.ast.value == prevState.raw && !this.props.ast.type
```

Comment:
What's this line meant to do?
`prevProps.ast.value` could sometimes be undefined